components/back-compatile-on-input/index.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+
+export default class BackCompatibleOnInput extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.onInput = this.onInput.bind( this );
+		this.onKeyUp = this.onKeyUp.bind( this );
+
+		this.state = {
+			isOnInputSupported: false,
+		};
+	}
+
+	onInput( event ) {
+		this.setState( {
+			onInputSupported: true,
+		} );
+		this.props.onInput( event );
+	}
+
+	onKeyUp( event ) {
+		this.props.onInput( event );
+	}
+
+	render() {
+		const { isOnInputSupported } = this.state;
+		const { onInput, children, ...props } = this.props;
+		return (
+			<div
```

Comment:
Wondering two options for avoiding the wrapping `div`:

- Apply as a `div`, but use in place of, not as a child, of the original component's `div` (i.e. the one with `className="components-autocomplete"`)
- Use `cloneElement` to apply the normalized event to the original child

Reply:
Not using a wrapper is challenging as the same component, may want a back-compatible onInput but also have an onKeyUp event (the case of autocomplete).
One option would be to use cloneElement but on the onKeyUp event pass a handler that does the logic of our component but also calls the existing onKeyUp event handler the component had.
I'm not certain if it the best way for this case, or if the additional complexity is valuable when compared to use a div wrapper.